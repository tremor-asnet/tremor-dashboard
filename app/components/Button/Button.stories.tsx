import type { Meta, StoryObj } from "@storybook/react";
import { Button, Text } from "@tremor/react";
import { TiHome } from "react-icons/ti";
import { RiArrowDropDownLine } from "react-icons/ri";

const meta = {
  title: "Components/Button",
  tags: ["autodocs"],
  component: Button,
} as Meta<typeof Button>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  render: () => (
    <Button className="font-bold bg-primary hover:bg-primary border-transparent text-xs">
      Primary
    </Button>
  ),
};

export const Secondary: Story = {
  render: () => (
    <Button
      className="font-bold bg-inherit text-tremor-content-seldom text-xs hover:bg-transparent hover:opacity-75"
      variant="secondary">
      Secondary
    </Button>
  ),
};

export const Light: Story = {
  render: () => (
    <Button
      className="font-bold text-tremor-content-title text-xs hover:text-tremor-content-title px-6 py-2.5"
      variant="light">
      Light
    </Button>
  ),
};
export const Icon: Story = {
  render: () => (
    <Button
      className="font-bold bg-inherit text-tremor-content-title border-0 border-inherit hover:bg-transparent text-xs"
      icon={TiHome}>
      App
    </Button>
  ),
};

export const Disable: Story = {
  render: () => (
    <Button
      disabled
      className="font-bold bg-primary hover:bg-primary border-transparent text-xs">
      Primary
    </Button>
  ),
};

export const NewOder: Story = {
  render: () => (
    <Button className="antialiased min-w-[64px] py-[12px] text-center uppercase sm:px-[21px] bg-gradient-primary dark:bg-gradient-pickled !shadow-btn-primary px-6 py-2.5 rounded-lg border-0 hover:!shadow-btn-primary-hover">
      <Text className="items-center uppercase py-[2px] text-xs font-bold font-primary text-white dark:text-dark-tremor-content-title">
        New Order
      </Text>
    </Button>
  ),
};

export const Filters: Story = {
  render: () => (
    <Button
      icon={RiArrowDropDownLine}
      iconPosition="right"
      className="antialiased flex items-center uppercase font-bold min-w-[64px] bg-transparent text-center border-primary focus:border-primary hover:border-primary focus:opacity-75 hover:opacity-75 uppercase sm:px-[26px] text-primary focus:text-white dark:text-dark-tremor-content-title hover:bg-transparent active:bg-primary focus:bg-primary dark:bg-gradient-pickled px-6 py-[9px] rounded-lg hover:!shadow-btn-primary-hover">
      <Text className="text-inherit text-xs">Filters</Text>
    </Button>
  ),
};
