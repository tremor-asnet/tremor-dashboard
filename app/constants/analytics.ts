/* Constants for Analytics Page */
export const SALE_STATISTICAL_TYPE = {
  REVENUE: "Revenue",
};
