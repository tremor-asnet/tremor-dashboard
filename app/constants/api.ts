import { ROUTER_API_URL } from "./routes";

export const API_ROUTES = {
  USERS: `${ROUTER_API_URL}/users`,
};
