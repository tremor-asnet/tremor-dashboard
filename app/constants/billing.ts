export const enum AGGREGATION_TYPE {
  SALARY = "Salary",
  PAYPAL = "Paypal",
}

export const enum AGGREGATION_DESCRIPTION {
  SALARY = "Belong Interactive",
  PAYPAL = "Freelance Payment",
}
