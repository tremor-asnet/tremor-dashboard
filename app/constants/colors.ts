type ColorChannelsChartType = {
  [key: number]: string;
  1: string;
  2: string;
  3: string;
  4: string;
};

export const CHANNELS_CHART_COLOR: ColorChannelsChartType = {
  1: "blue-500",
  2: "red-500",
  3: "primary",
  4: "gray-500",
};

export const TRANSACTION_COLOR = {
  INCREASE: "few",
  DECREASE: "attention",
  PENDING: "secondary",
};
