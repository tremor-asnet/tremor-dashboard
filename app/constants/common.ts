export const EXCEPT_KEYS = {
  POSITIVE_INTEGER: ["e", "E", "+", "-", "."],
  POSITIVE_DOUBLE: ["e", "E", "+", "-"],
};

export const DIRECTION = {
  ASC: "asc",
  DESC: "desc",
};
