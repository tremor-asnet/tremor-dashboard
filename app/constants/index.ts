export * from "./routes";
export * from "./regex";
export * from "./messages";
export * from "./profile";
export * from "./commons";
export * from "./flagSrc";
export * from "./images";
export * from "./sections";
export * from "./images";
export * from "./products";
export * from "./colors";
export * from "./units";
export * from "./sales";
export * from "./commons";
export * from "./status";
