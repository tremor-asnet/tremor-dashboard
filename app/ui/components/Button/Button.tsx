import { ElementType, ReactNode } from "react";

// Components
import {
  Button as ButtonTremor,
  ButtonVariant,
  HorizontalPosition,
  Size,
} from "@tremor/react";

// Constant
import { VARIANT_BUTTON } from "@/constants";

interface ButtonProps {
  variant?:
    | VARIANT_BUTTON.PRIMARY
    | VARIANT_BUTTON.PRIMARY_CENTER
    | VARIANT_BUTTON.SECONDARY
    | VARIANT_BUTTON.SECONDARY_SHADOW
    | VARIANT_BUTTON.LIGHT
    | VARIANT_BUTTON.LIGHT_CARD
    | VARIANT_BUTTON.LIGHT_STATUS
    | VARIANT_BUTTON.LIGHT_CENTER
    | VARIANT_BUTTON.SURFACE
    | VARIANT_BUTTON.DARK
    | "";
  variantTremor?: ButtonVariant;
  additionalClass?: string;
  icon?: ElementType;
  iconPosition?: HorizontalPosition;
  children?: string | ReactNode;
  disabled?: boolean;
  type?: "submit" | "reset" | "button";
  size?: Size;
  tabIndex?: number;
  onClick?: () => void;
  value?: string;
}

const Button = ({
  additionalClass = "",
  variant = "",
  variantTremor,
  icon,
  iconPosition,
  children,
  disabled,
  type,
  size,
  tabIndex,
  onClick,
  value,
  ...otherProps
}: ButtonProps) => {
  const renderTypeVariant = (variant: string) => {
    switch (variant) {
      case VARIANT_BUTTON.PRIMARY:
        return "bg-gradient-primary dark:bg-gradient-pickled";

      case VARIANT_BUTTON.SURFACE:
        return "items-start btn-form-secondary rounded-lg dark:bg-gradient-pickled py-3 px-6 mt-8 bg-gradient-btn-back hover:dark:!bg-gradient-pickled border-none dark:text-white text-center box-shadow-transparent";

      case VARIANT_BUTTON.PRIMARY_CENTER:
        return "btn-form-primary rounded-lg py-3 px-6 mt-8 bg-gradient-primary dark:bg-gradient-pickled hover:dark:!bg-gradient-pickled border-none dark:text-white text-center";

      case VARIANT_BUTTON.SECONDARY:
        return "!rounded-full border-secondary dark:border-secondary text-secondary hover:text-secondary hover:opacity-75 hover:bg-transparent p-[5.5px] box-shadow-transparent";

      case VARIANT_BUTTON.SECONDARY_SHADOW:
        return "box-shadow-transparent uppercase";

      case VARIANT_BUTTON.LIGHT_CARD:
        return "justify-start text-tremor-content-title hover:text-tremor-content-title hover:bg-body dark:hover:bg-dark-secondary hover:rounded-md";

      case VARIANT_BUTTON.LIGHT_CENTER:
        return "justify-center items-center opacity-50 dark:opacity-100";

      case VARIANT_BUTTON.LIGHT_STATUS:
        return "w-[25px] h-[25px] justify-center items-center rounded-full border border mr-2 text-tremor-content-title hover:text-tremor-content-title";

      case VARIANT_BUTTON.DARK:
        return "text-sm text-gray-400 dark:text-gray-400 bg-transparent hover:bg-select dark:bg-transparent dark:hover:greyish w-9 h-9 inline-flex justify-center items-center border-gray-300 hover:border-gray-300 !rounded-full box-shadow-transparent";

      default:
        return "";
    }
  };

  return (
    <ButtonTremor
      className={`${renderTypeVariant(variant)} ${additionalClass}`}
      variant={variantTremor}
      icon={icon}
      iconPosition={iconPosition}
      tabIndex={tabIndex}
      size={size}
      type={type}
      disabled={disabled}
      onClick={onClick}
      value={value}
      {...otherProps}>
      {children}
    </ButtonTremor>
  );
};

export default Button;
