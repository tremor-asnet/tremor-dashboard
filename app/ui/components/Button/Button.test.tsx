import { render, fireEvent } from "@testing-library/react";
import Button from "./Button";
import { VARIANT_BUTTON } from "@/constants";

describe("Button Component", () => {
  it("Should match snapshot", () => {
    const { container } = render(
      <div>
        <Button variant={VARIANT_BUTTON.PRIMARY}>Hello</Button>
        <Button variant={VARIANT_BUTTON.PRIMARY_CENTER}>Hello</Button>
        <Button variant={VARIANT_BUTTON.SECONDARY}>Hello</Button>
        <Button variant={VARIANT_BUTTON.SECONDARY_SHADOW}>Hello</Button>
        <Button variant={VARIANT_BUTTON.LIGHT}>Hello</Button>
        <Button variant={VARIANT_BUTTON.LIGHT_CARD}>Hello</Button>
        <Button variant={VARIANT_BUTTON.LIGHT_STATUS}>Hello</Button>
        <Button variant={VARIANT_BUTTON.LIGHT_CENTER}>Hello</Button>
        <Button variant={VARIANT_BUTTON.SURFACE}>Hello</Button>
        <Button variant={VARIANT_BUTTON.DARK}>Hello</Button>
        <Button variant={""}>Hello</Button>
      </div>,
    );
    expect(container).toMatchSnapshot();
  });

  it("Calls onClick callback when clicked", () => {
    const onClick = jest.fn();
    const { getByText } = render(<Button onClick={onClick}>Click Me</Button>);
    fireEvent.click(getByText("Click Me"));
    expect(onClick).toHaveBeenCalled();
  });
});
