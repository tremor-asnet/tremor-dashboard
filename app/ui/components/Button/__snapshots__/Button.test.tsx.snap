// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Should match snapshot 1`] = `
<div>
  <div>
    <button
      class="tremor-Button-root shrink-0 inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default border shadow-tremor-input dark:shadow-dark-tremor-input px-4 py-2 text-sm border-tremor-brand dark:border-dark-tremor-brand text-tremor-brand-inverted dark:text-dark-tremor-brand-inverted hover:bg-tremor-brand-emphasis dark:hover:bg-dark-tremor-brand-emphasis hover:border-tremor-brand-emphasis dark:hover:border-dark-tremor-brand-emphasis bg-gradient-primary dark:bg-gradient-pickled"
    >
      <span
        class="tremor-Button-text text-tremor-default whitespace-nowrap"
      >
        Hello
      </span>
    </button>
    <button
      class="tremor-Button-root shrink-0 inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default border shadow-tremor-input dark:shadow-dark-tremor-input text-sm border-tremor-brand dark:border-dark-tremor-brand text-tremor-brand-inverted hover:bg-tremor-brand-emphasis dark:hover:bg-dark-tremor-brand-emphasis hover:border-tremor-brand-emphasis dark:hover:border-dark-tremor-brand-emphasis btn-form-primary rounded-lg py-3 px-6 mt-8 bg-gradient-primary dark:bg-gradient-pickled hover:dark:!bg-gradient-pickled border-none dark:text-white text-center"
    >
      <span
        class="tremor-Button-text text-tremor-default whitespace-nowrap"
      >
        Hello
      </span>
    </button>
    <button
      class="tremor-Button-root shrink-0 inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default border shadow-tremor-input dark:shadow-dark-tremor-input text-sm bg-tremor-brand dark:bg-dark-tremor-brand dark:text-dark-tremor-brand-inverted dark:hover:bg-dark-tremor-brand-emphasis hover:border-tremor-brand-emphasis dark:hover:border-dark-tremor-brand-emphasis !rounded-full border-secondary dark:border-secondary text-secondary hover:text-secondary hover:opacity-75 hover:bg-transparent p-[5.5px] box-shadow-transparent"
    >
      <span
        class="tremor-Button-text text-tremor-default whitespace-nowrap"
      >
        Hello
      </span>
    </button>
    <button
      class="tremor-Button-root shrink-0 inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default border shadow-tremor-input dark:shadow-dark-tremor-input px-4 py-2 text-sm bg-tremor-brand dark:bg-dark-tremor-brand border-tremor-brand dark:border-dark-tremor-brand text-tremor-brand-inverted dark:text-dark-tremor-brand-inverted hover:bg-tremor-brand-emphasis dark:hover:bg-dark-tremor-brand-emphasis hover:border-tremor-brand-emphasis dark:hover:border-dark-tremor-brand-emphasis box-shadow-transparent uppercase"
    >
      <span
        class="tremor-Button-text text-tremor-default whitespace-nowrap"
      >
        Hello
      </span>
    </button>
    <button
      class="tremor-Button-root shrink-0 inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default border shadow-tremor-input dark:shadow-dark-tremor-input px-4 py-2 text-sm bg-tremor-brand dark:bg-dark-tremor-brand border-tremor-brand dark:border-dark-tremor-brand text-tremor-brand-inverted dark:text-dark-tremor-brand-inverted hover:bg-tremor-brand-emphasis dark:hover:bg-dark-tremor-brand-emphasis hover:border-tremor-brand-emphasis dark:hover:border-dark-tremor-brand-emphasis"
    >
      <span
        class="tremor-Button-text text-tremor-default whitespace-nowrap"
      >
        Hello
      </span>
    </button>
    <button
      class="tremor-Button-root shrink-0 inline-flex items-center group font-medium outline-none rounded-tremor-default border shadow-tremor-input dark:shadow-dark-tremor-input px-4 py-2 text-sm bg-tremor-brand dark:bg-dark-tremor-brand border-tremor-brand dark:border-dark-tremor-brand dark:text-dark-tremor-brand-inverted hover:border-tremor-brand-emphasis dark:hover:border-dark-tremor-brand-emphasis justify-start text-tremor-content-title hover:text-tremor-content-title hover:bg-body dark:hover:bg-dark-secondary hover:rounded-md"
    >
      <span
        class="tremor-Button-text text-tremor-default whitespace-nowrap"
      >
        Hello
      </span>
    </button>
    <button
      class="tremor-Button-root shrink-0 inline-flex group font-medium outline-none rounded-tremor-default shadow-tremor-input dark:shadow-dark-tremor-input px-4 py-2 text-sm bg-tremor-brand dark:bg-dark-tremor-brand border-tremor-brand dark:border-dark-tremor-brand dark:text-dark-tremor-brand-inverted hover:bg-tremor-brand-emphasis dark:hover:bg-dark-tremor-brand-emphasis hover:border-tremor-brand-emphasis dark:hover:border-dark-tremor-brand-emphasis w-[25px] h-[25px] justify-center items-center rounded-full border mr-2 text-tremor-content-title hover:text-tremor-content-title"
    >
      <span
        class="tremor-Button-text text-tremor-default whitespace-nowrap"
      >
        Hello
      </span>
    </button>
    <button
      class="tremor-Button-root shrink-0 inline-flex group font-medium outline-none rounded-tremor-default border shadow-tremor-input dark:shadow-dark-tremor-input px-4 py-2 text-sm bg-tremor-brand dark:bg-dark-tremor-brand border-tremor-brand dark:border-dark-tremor-brand text-tremor-brand-inverted dark:text-dark-tremor-brand-inverted hover:bg-tremor-brand-emphasis dark:hover:bg-dark-tremor-brand-emphasis hover:border-tremor-brand-emphasis dark:hover:border-dark-tremor-brand-emphasis justify-center items-center opacity-50 dark:opacity-100"
    >
      <span
        class="tremor-Button-text text-tremor-default whitespace-nowrap"
      >
        Hello
      </span>
    </button>
    <button
      class="tremor-Button-root shrink-0 inline-flex justify-center group font-medium outline-none rounded-tremor-default border shadow-tremor-input dark:shadow-dark-tremor-input text-sm border-tremor-brand dark:border-dark-tremor-brand text-tremor-brand-inverted hover:bg-tremor-brand-emphasis dark:hover:bg-dark-tremor-brand-emphasis hover:border-tremor-brand-emphasis dark:hover:border-dark-tremor-brand-emphasis items-start btn-form-secondary rounded-lg dark:bg-gradient-pickled py-3 px-6 mt-8 bg-gradient-btn-back hover:dark:!bg-gradient-pickled border-none dark:text-white text-center box-shadow-transparent"
    >
      <span
        class="tremor-Button-text text-tremor-default whitespace-nowrap"
      >
        Hello
      </span>
    </button>
    <button
      class="tremor-Button-root shrink-0 group font-medium outline-none rounded-tremor-default border shadow-tremor-input dark:shadow-dark-tremor-input px-4 py-2 dark:border-dark-tremor-brand dark:hover:bg-dark-tremor-brand-emphasis dark:hover:border-dark-tremor-brand-emphasis text-sm text-gray-400 dark:text-gray-400 bg-transparent hover:bg-select dark:bg-transparent dark:hover:greyish w-9 h-9 inline-flex justify-center items-center border-gray-300 hover:border-gray-300 !rounded-full box-shadow-transparent"
    >
      <span
        class="tremor-Button-text text-tremor-default whitespace-nowrap"
      >
        Hello
      </span>
    </button>
    <button
      class="tremor-Button-root shrink-0 inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default border shadow-tremor-input dark:shadow-dark-tremor-input px-4 py-2 text-sm bg-tremor-brand dark:bg-dark-tremor-brand border-tremor-brand dark:border-dark-tremor-brand text-tremor-brand-inverted dark:text-dark-tremor-brand-inverted hover:bg-tremor-brand-emphasis dark:hover:bg-dark-tremor-brand-emphasis hover:border-tremor-brand-emphasis dark:hover:border-dark-tremor-brand-emphasis"
    >
      <span
        class="tremor-Button-text text-tremor-default whitespace-nowrap"
      >
        Hello
      </span>
    </button>
  </div>
</div>
`;
